From: "Torne (Richard Coles)" <torne@google.com>
Date: Fri, 22 Jul 2022 18:22:34 +0000
Subject: Clean up logic in BuildInfo.targetsAtLeastT.

The logic previously implemented worked correctly, but was misleading as
it implied that there were actually two separate cases for pre and post
API finalization, when actually only the first case matters because
CUR_DEVELOPMENT is greater than the actual API level.

Clean this up and tweak the comments to make it clearer that the
transition from API finalization to public SDK will be fine and the only
reason for needing three "versions" of the function here is because the
named constant won't be available at build time upstream yet.

Bug: 1339376
Change-Id: I9090300cf098753f02b9f13770fba51b8d76d388
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/3782717
Reviewed-by: Jinsuk Kim <jinsukkim@chromium.org>
Commit-Queue: Jinsuk Kim <jinsukkim@chromium.org>
Auto-Submit: Richard Coles <torne@chromium.org>
Commit-Queue: Richard Coles <torne@chromium.org>
Cr-Commit-Position: refs/heads/main@{#1027336}
---
 .../java/src/org/chromium/base/BuildInfo.java    | 16 +++++++++-------
 1 file changed, 9 insertions(+), 7 deletions(-)

diff --git a/base/android/java/src/org/chromium/base/BuildInfo.java b/base/android/java/src/org/chromium/base/BuildInfo.java
--- a/base/android/java/src/org/chromium/base/BuildInfo.java
+++ b/base/android/java/src/org/chromium/base/BuildInfo.java
@@ -248,13 +248,15 @@ public class BuildInfo {
         // Logic for pre-API-finalization:
         // return BuildCompat.isAtLeastT() && target == Build.VERSION_CODES.CUR_DEVELOPMENT;
 
-        // Logic for smooth transition period so that both pre-finalization and final SDKs
-        // will return true, assuming T will be API 33.
-        // Keeping this until we upstream the public SDK is a reasonable transition period.
-        return target >= 33 ||
-                (BuildCompat.isAtLeastT() && target == Build.VERSION_CODES.CUR_DEVELOPMENT);
-
-        // Logic for public SDK release:
+        // Logic for after API finalization but before public SDK release has to
+        // just hardcode the appropriate SDK integer. This will include Android
+        // builds with the finalized SDK, and also pre-API-finalization builds
+        // (because CUR_DEVELOPMENT == 10000).
+        return target >= 33;
+
+        // Once the public SDK is upstreamed we can use the defined constant,
+        // deprecate this, then eventually inline this at all callsites and
+        // remove it.
         // return target >= Build.VERSION_CODES.TIRAMISU;
     }
 
-- 
2.37.3

